lib/package.py

```diff
@@ -180,3 +184,31 @@ def _download_build_files(self):
             filename = os.path.join(self.build_files, url.split('/')[-1])
             with open(filename, "wb") as file_data:
                 file_data.write(data)
+
+    def lock(self):
+        """
+        Locks the package to avoid concurrent operations on its shared
+        resources.
+        Currently, the only resource shared among scripts executed from
+        different directories is the repository.
+        """
+        LOG.debug("Checking for lock on file {}.".format(self.lock_file_path))
+        self.lock_file = open(self.lock_file_path, "w")
+        try:
+            fcntl.lockf(self.lock_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
What if `lock_file` is locked, i.e. opened by another process?  Shall we use any busy-waiting technique to wait until it can get the lock?